.clang-tidy

```diff
@@ -0,0 +1,12 @@
+Checks:
+  *,
+  -altera*,
+  -llvm*,
+  -fuchsia-default-arguments-*,
+  -bugprone-easily-swappable-parameters,
+  -cppcoreguidelines-avoid-magic-numbers,
+  -modernize-use-trailing-return-type,
+  -readability-magic-numbers,
+CheckOptions:
+  - { key: hicpp-signed-bitwise.IgnorePositiveIntegerLiterals, value: 1 }
+  - { key: misc-non-private-member-variables-in-classes.IgnoreClassesWithAllMemberVariablesBeingPublic, value: 1 }
```

Comment:
I'd be fine with this check (i.e. a `false` value). If this is the case, why don't we just use a struct?

### Reference
[1] https://clang.llvm.org/extra/clang-tidy/checks/misc/non-private-member-variables-in-classes.html

Reply:
The check doesn't differentiate between classes and structs so even structs with all public members get hit by the check if we don't have this option enabled. Which is understandable, because C++ says they're the same (except for default visibility).

Reply:
I agree with this if it bugs you about all variables being public in a struct. That's what structs are for lol